app/modules/users/models.py

```diff
@@ -526,6 +527,43 @@ def has_permission_to_read(self, obj):
 
         return has_permission
 
+    # Permissions control entry point for real users, for all objects and all operations
```

Comment:
Feedback request.
I put these two functions here as it seemed best to have the permissions logic together.
Having had a thought about the role handling, I now think this logic should move into the permissions/roles.py
Anyone agree/disagree?
If so, should this be reworked or taken as is and changed later?
I guess it depends on if Colin is waiting for these changes.

Reply:
Let's move the check for `obj` being `None` to the top of the function as 

```
if obj is None:
    return True
```

As for the check for "clone if not known" on submissions, we already account for this by returning a `None` for `obj` value and we further check if the repository can be cloned.  In this way, we are bypassing the checks for HTTP `HTTPStatus.NOT_FOUND` error with a failed database look up using the `return_not_found` argument (see https://github.com/WildbookOrg/houston/blob/develop/app/modules/submissions/resources.py#L150)

Reply:
Functionality updated so that this code (now in rules.py) only handles valid objects (and assert this)  in the check in ObjectActionRule. For calls where there is no object, ModuleActionRule is to be used